test/ScheduledChannelTest.cpp

```diff
@@ -0,0 +1,112 @@
+#include "pipert/ScheduledChannel.h"
+#include "pipert/Scheduler.h"
+#include "gtest/gtest.h"
+
+namespace{
+    class OtherPrinter {
+        public:
+        void Print(pipert::PacketToProcess<int> packet) {
+            std::cout << "Other: " << packet.data() << std::endl;
```

Comment:
The whole structure and approach is good here: for testing, you need to make a side-effect. However, console output is something you cannot test for. On the other hand, it litters the console with garbage. If the tests are run parallel by the unit test framework, the messages will even be mixed.
I think, you can skip the console output and do something testable instead. Setting a variable in a Packet like down here is a good approach. Setting a global variable or a statis one in the anonymous namespace can also work.

Reply:
I have changed the code accordingly. (I am commenting here, in case there is no notification for my last commit)